Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -900,6 +900,13 @@ void Painter::blit_dimmed(const IntPoint& position, const Gfx::Bitmap& source, c
     });
 }
 
+void Painter::blit_grayscaled(const IntPoint& position, const Gfx::Bitmap& source, const IntRect& src_rect)
+{
+    return blit_filtered(position, source, src_rect, [](Color src) {
+        return src.to_grayscale();
+    });
+}
+
```

Comment:
Is it necessary to add a grayscale blit specialization? If we actually need the grayscale filter a lot, it might be better to provide a couple of common filter lambdas like the one you have here.

Reply:
Yes, I agree with that. I think adding filter lambdas is outside of this PR though.